From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nils Kuhnhenn <kuhnhenn.nils@gmail.com>
Date: Wed, 24 Aug 2016 19:45:02 +0200
Subject: [PATCH 4/7] ntdll/heap.c: freelist_balance - prefetch next entry
 cache line

---
 dlls/ntdll/heap.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/dlls/ntdll/heap.c b/dlls/ntdll/heap.c
index d59f3c8414..57d3d544e5 100644
--- a/dlls/ntdll/heap.c
+++ b/dlls/ntdll/heap.c
@@ -595,6 +595,7 @@ static DWORD freelist_balance ( HEAP *heap, unsigned int freelistIndex )
     while (1)
     {
         ptr = ptr->next;
+        __builtin_prefetch(ptr->next, 0, 1);
 
         pArena = LIST_ENTRY( ptr, ARENA_FREE, entry );
 
-- 
2.12.0

