From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nils Kuhnhenn <kuhnhenn.nils@gmail.com>
Date: Thu, 27 Apr 2017 19:21:23 +0200
Subject: [PATCH 7/7] wined3d: use SwitchToThread() in wined3d_pause();

---
 dlls/wined3d/wined3d_private.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index 0dd61e71a3..5b248fafe7 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -365,7 +365,8 @@ static inline unsigned int wined3d_popcount(unsigned int x)
 static inline void wined3d_pause(void)
 {
 #if defined(__GNUC__) && (defined(__i386__) || defined(__x86_64__))
-    __asm__ __volatile__( "rep;nop" : : : "memory" );
+    SwitchToThread();
+    //__asm__ __volatile__( "rep;nop" : : : "memory" );
 #endif
 }
 
-- 
2.12.2

